frontend/packages/integration-tests-cypress/tests/app/demo-dynamic-plugin.spec.ts

```diff
@@ -0,0 +1,188 @@
+import { safeDump, safeLoadAll } from 'js-yaml';
+import * as _ from 'lodash';
+import { checkErrors } from '../../support';
+import { detailsPage } from '../../views/details-page';
+import {
+  infoMessage,
+  hintMessage,
+  warningMessage,
+  warningToast,
+  errorMessage,
+} from '../../views/form';
+import { listPage } from '../../views/list-page';
+import { modal } from '../../views/modal';
+import { nav } from '../../views/nav';
+import * as yamlEditor from '../../views/yaml-editor';
+
+const PLUGIN_NAME = 'console-demo-plugin';
+const PLUGIN_PATH = '../../../dynamic-demo-plugin';
+const CHECK_UPDATE_WAIT = 30000;
+const PLUGIN_PULL_SPEC = Cypress.env('PLUGIN_PULL_SPEC');
+
+/*
+  These tests are meant to:
+    1. show how to test a dynamic plugin using demo as the plugin instance
+    2. run locally:
+      2a. build the plugin locally, and run the server
+      2b. using bridge running with the plugin argument that points to the local dynanmic plugin server
+      2c. will intercept the check update call to mock a toast notification that a plugin has been enabled
+      2d. will not use all workload definitions defined in the yaml (not using the env variable for pull spec)
+    3. run on ci:
+      3a. ci will build the dynamic plugin and provide the pullspec in the env var: CYPRESS_PLUGIN_PULL_SPEC
+      3b. that pull spec will be used to create the deployment on the cluster
+      3c. intercepting the check-updates call will not occur as the toast will appear when the plugin is enabled
+    4. the scaffolding should remain the same except modifying the constants above
+ */
+
+const enableDemoPlugin = (enable: boolean) => {
+  // find console demo plugin and enable it
+  cy.visit('k8s/cluster/operator.openshift.io~v1~Console/cluster/console-plugins');
+  cy.url().should(
+    'include',
+    'k8s/cluster/operator.openshift.io~v1~Console/cluster/console-plugins',
+  );
+  cy.get('.co-resource-item__resource-name')
+    .byLegacyTestID(PLUGIN_NAME)
+    .should('be.visible');
+  cy.byLegacyTestID(PLUGIN_NAME)
+    .parents('tr')
+    .within(() => {
+      cy.byTestID('edit-console-plugin').contains(enable ? 'Disabled' : 'Enabled');
+      cy.byTestID('edit-console-plugin').click();
+    });
+  modal.shouldBeOpened();
+  cy.contains('Cancel');
+  modal.modalTitleShouldContain('Console plugin enablement');
+  cy.byTestID(enable ? 'Enable-radio-input' : 'Disable-radio-input').click();
+  modal.submit();
+  modal.shouldBeClosed();
+  cy.byLegacyTestID(PLUGIN_NAME)
+    .parents('tr')
+    .within(() => {
+      cy.byTestID('edit-console-plugin').contains(enable ? 'Enabled' : 'Disabled');
+    });
+  cy.window().then((win: any) => {
+    if (win.SERVER_FLAGS?.authDisabled) {
```

Comment:
This isn't really the correct check for a local console. It's possible to run locally with auth enabled as an example.

Reply:
@spadgett @dtaylor113 would this be the correct test for a local bridge env?

`if (!Cypress.config('baseUrl').includes('localhost')) `

Reply:
Added a common way to test for local bridge execution:
`export const isLocalDevEnvironment = Cypress.config('baseUrl').includes('localhost');`